katdal/chunkstore_s3.py

```diff
@@ -244,6 +245,11 @@ def decode_jwt(token):
         # This covers e.g. bad characters in the signature or non-JSON-dict payload
         raise_from(InvalidToken(token, "Could not decode token - maybe it's truncated "
                                        "or corrupted? ({})".format(err)), err)
+    except jwt.exceptions.ExpiredSignatureError as err:
+        claims = jwt.decode(token, verify=False)
```

Comment:
I haven't looked through the sequence of operations in pyjwt; is there any risk that this will also raise an exception e.g. if the signature has expired AND there is something else wrong, and turning off verification exposes the second problem. I'm guessing it should be okay.

Reply:
I did check that. At the JWT level `verify=False` lets `pyjwt` load the JWS segments (while not verifying the signature) and then extract the JSON payload without any further validation of claims.

My only concern is whether the older `verify` parameter is deprecated in favour of `options`. It seems to be discouraged at the JWS level but still accepted at the JWT level. And it is very convenient for this use case.